Date feature by Hansen

HansenLudwig opened the pull request.

HansenLudwig committed:
Dummy Cmt

HansenLudwig committed:
modified:   webapp/controller/App.controller.js
	new file:   webapp/i18n/i18n_zh.properties
	modified:   webapp/manifest.json
	modified:   webapp/model/todoitems.json
	modified:   webapp/view/App.view.xml

HansenLudwig committed:
On branch DateFeature
 Your branch is up to date with 'origin/DateFeature'.

 Changes to be committed:
	modified:   webapp/Component.js
	modified:   webapp/controller/App.controller.js
	modified:   webapp/css/styles.css
	modified:   webapp/i18n/i18n.properties
	modified:   webapp/i18n/i18n_de.properties
	modified:   webapp/i18n/i18n_en.properties
	modified:   webapp/i18n/i18n_zh.properties
	modified:   webapp/manifest.json
	modified:   webapp/model/todoitems.json
	modified:   webapp/view/App.view.xml

HansenLudwig committed:
modified:   webapp/view/App.view.xml

HansenLudwig committed:
modified:   webapp/manifest.json

HansenLudwig committed:
Merge branch 'DateFeature' of https://github.com/HansenLudwig/openui5-sample-app into DateFeature

HansenLudwig committed:
modified:   webapp/controller/App.controller.js
        HashTag ver 1.0

HansenLudwig committed:
[Warning] it's NOT running correct!

 Changes to be committed:
	modified:   webapp/controller/App.controller.js
        Seems like sth. wrong with the js file.
	modified:   webapp/model/todoitems.json
        Structure of .json has been changed, all the compo. and
      views should be attached to it again.
	modified:   webapp/view/App.view.xml
        worked together with Fab and Stef, had some useful changes.

  Stil a lot of work to do!

HansenLudwig committed:
Changes to be committed:
	modified:   webapp/controller/App.controller.js
	modified:   webapp/css/styles.css
	modified:   webapp/i18n/i18n.properties
	modified:   webapp/i18n/i18n_de.properties
	modified:   webapp/i18n/i18n_en.properties
	modified:   webapp/i18n/i18n_zh.properties
	modified:   webapp/manifest.json
	new file:   webapp/model/formatter.js
	modified:   webapp/model/todoitems.json
	modified:   webapp/view/App.view.xml

Before A huge to manifest.json.....

HansenLudwig committed:
DateOfAdd: Fixed
DDL: Fixed
SearchOnTitleAndHashTag: somehow working
Priority: Looking Good

HansenLudwig committed:
modified:   webapp/controller/App.controller.js

HansenLudwig committed:
Merge branch 'DateFeature' of https://github.com/HansenLudwig/openui5-sample-app into DateFeature

fabiantschirschnitz commented on webapp/Component.js:
```diff
@@ -1,7 +1,11 @@
+
 	return UIComponent.extend("sap.ui.demo.todo.Component", {
 		metadata: {
+			// interfaces: ["sap.ui.core.IAsyncContentCreation"],
```
why was this added?

HansenLudwig replied:
![image](https://github.com/HansenLudwig/openui5-sample-app/assets/60106776/dfeb1cc3-6f19-4536-8c70-826cc44a45f9)
Hi, we can check if it can be seen after this Pull Req..
As shown in the Pic, I deleted this comment and added a new comment here.
 - VG

fabiantschirschnitz commented on webapp/controller/App.controller.js:
```diff
@@ -3,36 +3,105 @@ sap.ui.define([
 
 		onInit() {
+
+			let oModel = new JSONModel();
```
why let?

fabiantschirschnitz commented on webapp/controller/App.controller.js:
```diff
@@ -3,36 +3,105 @@ sap.ui.define([
+
+			let oModel = new JSONModel();
+			
+			let oOptions = {
```
same

fabiantschirschnitz commented on webapp/Component.js:
```diff
@@ -1,7 +1,11 @@
-sap.ui.define(["sap/ui/core/UIComponent", "sap/ui/core/ComponentSupport"], (UIComponent) => {
+sap.ui.define(["sap/ui/core/UIComponent", 
+	"sap/ui/core/ComponentSupport"
+], (UIComponent, ComponentSupport) => {
```
did you use repo formatting (e.g. prettier, etc.)

fabiantschirschnitz commented on webapp/controller/App.controller.js:
```diff
@@ -3,36 +3,105 @@ sap.ui.define([
+			let oModel = new JSONModel();
+			
+			let oOptions = {
+				"SelectedOption": "Normal(Default)",
```
this can be removed

fabiantschirschnitz commented on webapp/controller/App.controller.js:
```diff
@@ -3,36 +3,105 @@ sap.ui.define([
+					{
+						"PriorityID": "5",
+						"PriName": "Top",
+						"Icon": "sap-icon://add-product"
```
why the use all the same icon

fabiantschirschnitz commented on webapp/controller/App.controller.js:
```diff
@@ -3,36 +3,105 @@ sap.ui.define([
+
+			this.getView().setModel(oModel, "view");
+			// should do it on component level.
+			this.getView().getModel().setProperty("/default_newTodo/DDLAtUTC", JSON.stringify(UI5Date.getInstance()).slice(1,-1))
```
you could add this string to the todoitems.json

fabiantschirschnitz commented on webapp/controller/App.controller.js:
```diff
@@ -3,36 +3,105 @@ sap.ui.define([
 			const oModel = this.getView().getModel();
+			if(!oModel.getProperty("/newTodo/title"))
+			{
+				return
```
add some error message to the user? or even better form validation?!

fabiantschirschnitz commented on webapp/controller/App.controller.js:
```diff
@@ -3,36 +3,105 @@ sap.ui.define([
+				title: newTodoTitle,
+				hashTag: newTodoHashTag,
+				completed: false,
+				priority: newTodo.priority,  //newTodoPriorityIndex[newTodo.priority],
```
can this be removed?

fabiantschirschnitz commented on webapp/controller/App.controller.js:
```diff
@@ -3,36 +3,105 @@ sap.ui.define([
 
 			oModel.setProperty("/todos", aTodos);
-			oModel.setProperty("/newTodo", "");
+			oModel.setProperty("/newTodo", defaultNewTodo);
```
maybe a nice comment that you reset the model with a new semi-empty todo would be nice here :)

fabiantschirschnitz reviewed (changes requested):
lgtm. some minor changes requested so far. :)

HansenLudwig committed:
Changes to be committed:
	modified:   webapp/Component.js
        added a comment to test if it can be seen after PullReq.

HansenLudwig committed:
Merge branch 'DateFeature' of https://github.com/HansenLudwig/openui5-sample-app into DateFeature

HansenLudwig committed:
new file:   webapp/test/initMockServer.js
	new file:   webapp/test/mockServer.html
